cmake/Open3DSetGlobalProperties.cmake

```diff
@@ -5,6 +5,11 @@ function(open3d_set_global_properties target)
     # Tell CMake we want a compiler that supports C++14 features
     target_compile_features(${target} PUBLIC cxx_std_14)
 
+    # std::filesystem (C++17) or std::experimental::filesystem (C++14)
+    if(UNIX AND NOT APPLE)
+        target_link_libraries(${target} PRIVATE stdc++fs)
```

Comment:
Is this a compiler specific decision instead of OS specific? I see on cppreference:
`Using this library may require additional compiler/linker options. GNU implementation requires linking with -lstdc++fs and LLVM implementation requires linking with -lc++fs.`